likecoin/js/admin-settings/src/components/MainSettingTable.js

```diff
@@ -0,0 +1,248 @@
+import {
+  useRef, useState, useEffect, useMemo,
+} from 'react';
+import axios from 'axios';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { debounce } from 'lodash';
+import SubmitButton from './SubmitButton';
+import CheckBox from './CheckBox';
+import DropDown from './DropDown';
+import Section from './Section';
+import LikecoinInfoTable from './LikecoinInfoTable';
+import SettingNotice from './SettingNotice';
+import LikecoinHeading from './LikecoinHeading';
+import { SITE_LIKER_INFO_STORE_NAME } from '../store/site-likerInfo-store';
+
+function MainSettingTable(props) {
+  /* do not want to re-render the whole component until submit. Hence use useRef(). */
+  const siteLikerIdEnabledRef = useRef();
+  const displayOptionRef = useRef();
+  const perPostOptionEnabledRef = useRef();
+
+  const {
+    DBSiteLikerId,
+    DBSiteLikerAvatar,
+    DBSiteLikerDisplayName,
+    DBSiteLikerWallet,
+    DBSiteLikerIdEnabled,
+    DBDisplayOptionSelected,
+    DBPerPostOptionEnabled,
+  } = useSelect((select) => ({
+    DBSiteLikerId: select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo()
+      .DBSiteLikerId,
+    DBSiteLikerAvatar: select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo()
+      .DBSiteLikerAvatar,
+    DBSiteLikerDisplayName: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBSiteLikerDisplayName,
+    DBSiteLikerWallet: select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo()
+      .DBSiteLikerWallet,
+    DBSiteLikerIdEnabled: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBSiteLikerIdEnabled,
+    DBDisplayOptionSelected: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBDisplayOptionSelected,
+    DBPerPostOptionEnabled: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBPerPostOptionEnabled,
+  }));
```

Comment:
Is it equivalent to this? 
```suggestion
  } = useSelect((select) => select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo());
```

Reply:
true! much clear, thanks